### PR TITLE
[refactor](fs)(step1) add new storage file system

### PR DESCRIPTION
**Pull request by wsjz**
# Proposed changes

see https://github.com/apache/doris/issues/18960

PR1:  add new storage file system template and move old storage to new package
PR2: extract some method in old storage to new file system.
PR3: use storages  to access remote object storage, and use  file systems to access file in local or remote location. Will add some unit tests.

this is PR1.

image

add FileOperations in PR2

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/fs/obj/HdfsStorage.java [hunk 1]
```diff
@@ -15,9 +15,11 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package org.apache.doris.backup;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/fs/obj/HdfsStorage.java:18]: Why moving `hdfs` to `fs/obj`? hdfs is not a object storage system

Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/fs/obj/HdfsStorage.java:18]: I will remove it in PR2 as it is a deprecated class.

**Comment by wsjz** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**